swiftnav/src/signal/code.rs

```diff
@@ -0,0 +1,616 @@
+// Copyright (c) 2025 Swift Navigation Inc.
+// Contact: Swift Navigation <dev@swiftnav.com>
+//
+// This source is subject to the license found in the file 'LICENSE' which must
+// be be distributed together with this source. All other rights reserved.
+//
+// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+use super::{consts, Constellation};
+
+/// Code identifiers
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    Ord,
+    PartialOrd,
+    Eq,
+    PartialEq,
+    Hash,
+    strum::AsRefStr,
+    strum::Display,
+    strum::EnumIter,
```

Comment:
It bothers me that this functionality still has not landed in serde yet as it seems like it would be pretty straightfoward to add and many projects end up using both serde and strum